pkg/mutation/mutate.go

```diff
@@ -116,3 +132,225 @@ func UpdateMutatedContentWithComments(yamlContent string, comments []config.Muta
 	}
 	return fileContent
 }
+
+func findNodes(node *yaml.Node, selectors []string) ([]*yaml.Node, error) {
```

Comment:
This function looks great!

I wonder if we can break it up a bit to make it a little more readable...some comments below